ingestion/src/metadata/ingestion/source/database/processor.py

```diff
@@ -0,0 +1,166 @@
+        if regex_result.credit_cards:  # pylint: disable=no-member
+            types.append(PiiTypes.CREDIT_CARD.name)
+        if regex_result.street_addresses:  # pylint: disable=no-member
+            types.append(PiiTypes.ADDRESS.name)
+
+        return types
+
+
+class ColumnNameScanner(Scanner):
+    """
+    Column Name Scanner to scan column name
+    """
+
+    regex = {
+        PiiTypes.PERSON: re.compile(
+            "^.*(firstname|fname|lastname|lname|"
+            "fullname|maidenname|_name|"
+            "nickname|name_suffix|name).*$",
+            re.IGNORECASE,
+        ),
+        PiiTypes.EMAIL: re.compile("^.*(email|e-mail|mail).*$", re.IGNORECASE),
+        PiiTypes.BIRTH_DATE: re.compile(
+            "^.*(date_of_birth|dateofbirth|dob|"
+            "birthday|date_of_death|dateofdeath).*$",
+            re.IGNORECASE,
+        ),
+        PiiTypes.GENDER: re.compile("^.*(gender).*$", re.IGNORECASE),
+        PiiTypes.NATIONALITY: re.compile("^.*(nationality).*$", re.IGNORECASE),
+        PiiTypes.ADDRESS: re.compile(
+            "^.*(address|city|state|county|country|"
+            "zipcode|zip|postal|zone|borough).*$",
+            re.IGNORECASE,
+        ),
+        PiiTypes.USER_NAME: re.compile("^.*user(id|name|).*$", re.IGNORECASE),
+        PiiTypes.PASSWORD: re.compile("^.*pass.*$", re.IGNORECASE),
+        PiiTypes.SSN: re.compile("^.*(ssn|social).*$", re.IGNORECASE),
+        PiiTypes.PHONE: re.compile("^.*(phone).*$", re.IGNORECASE),
+        PiiTypes.KEY: re.compile("^.*(key).*$", re.IGNORECASE),
+    }
+
+    def scan(self, text):
+        types = set()
+        for pii_type_keys, pii_type_pattern in self.regex.items():
+            if pii_type_pattern.match(text) is not None:
+                types.add(pii_type_keys)
+
+        logging.debug("PiiTypes are %s", ",".join(str(x) for x in list(types)))
+        return list(types)
+
+
+class PiiProcessor(Processor):
+    """
+    Processor class to process columns of table
+    """
+
+    metadata_config: OpenMetadata
+    status: ProcessorStatus
+    metadata: OpenMetadata
+
+    def __init__(self, metadata_config: OpenMetadata):
+        super().__init__()
+        self.metadata = metadata_config
+        self.status = ProcessorStatus()
+        self.column_scanner = ColumnNameScanner()
+
+    @classmethod
+    def create(cls, config_dict: dict):  # pylint: disable=arguments-differ
+        metadata_config = OpenMetadataConnection.parse_obj(config_dict)
+        return cls(metadata_config)
+
+    def process(  # pylint: disable=arguments-differ
+        self, table_request: CreateTableRequest
+    ) -> Optional[CreateTableRequest]:
+        for column in table_request.columns:
+            pii_tags = []
+            pii_tags += self.column_scanner.scan(column.name.__root__)
+            tag_labels = []
+            if pii_tags:
+                tag_labels.append(
+                    TagLabel(
+                        tagFQN="PII.Sensitive",
```

Comment:
Let's set the Tag FQN as constants at the top. For the `labelType`, `state` and `source` I believe we already have pydantic models as Enums where we can get that listed instead of writing the string directly

Reply:
Ok 👍